iina/ObjcUtils.m

```diff
@@ -70,7 +77,7 @@ + (NSUInteger)levDistance:(NSString *)str0 and:(NSString *)str1 {
     for (i = 1; i <= len0; ++i)
       (*dist)[i][j] = min((*dist)[i - 1][j] + INDEL_WEIGHT,
                           (*dist)[i][j - 1] + INDEL_WEIGHT,
-                          (*dist)[i - 1][j - 1] + (cstr0[i] == cstr1[j] ? 0 : SUBSTITUTION_WEIGHT));
+                          (*dist)[i - 1][j - 1] + (cstr0[i - 1] == cstr1[j - 1] ? 0 : SUBSTITUTION_WEIGHT));
```

Comment:
This is not off-by-one error. This is part of the state transition equation (`dist[i][j]` is updated by `dist[i - 1][j - 1]` and the cost at (i, j))

Reply:
Been too long since I worked on this so I'm still trying to remember clearly, plus a bit burned out tonight, however I believe this is the issue:

`dist` is allocated like so with len + 1:
```swift
  int *_dist = malloc(sizeof(int) * (len0 + 1) * (len1 + 1));
```

The indexes start at 1 and go to len:
```swift
  for (j = 1; j <= len1; ++j)
    for (i = 1; i <= len0; ++i)
```

This all good for accessing `dist` with `[i - 1][j - 1]`.

But this code is accessing the 0 based strings with a 1 based index:
```swift
(*dist)[i - 1][j - 1] + (cstr0[i] == cstr1[j] ? 0 : SUBSTITUTION_WEIGHT));
```

That means it skips the over the first character in the two strings since `i` and `j` start at 1. That seems very wrong.

And since the for loop goes to `<= len` when the index gets to the length of the string the index is out of bounds, triggering the address sanitizer. 

I think the change is needed and correct? At least there is no question something is wrong here as I just confirmed that the address sanitizer triggers if I back out that change.

I am going to push a rebase with develop to get the CI build to pass.

Reply:
> But this code is accessing the 0 based strings with a 1 based index:

On line 51 & 52, I added a space before the 2 strings. Then, in the transition equation, we have already covered all 3 initial possibilities (0, 1) (1, 0) (1, 1). There's no reason for (0, 0) since the 0th character is the space.

> And since the for loop goes to <= len when the index gets to the length of the string the index is out of bounds, triggering the address sanitizer.

This is because previously, the strings are null terminated (C-style strings), thus accessing the len'th character won't cause any issue. Since you have changed it to non null-terminated, you should add another character ('\0' is a good candidate) at the end of each strings.